### PR TITLE
Feat/ optimize profile menu

### PR DESCRIPTION
**Pull request by langemike**
This change is required because of two reasons:
1. Since the introduction of the UserMenu heading, it caused "Settings" to displayed before the profile menu buttons. This bug is introduced in PR #120
2. The profile menu buttons also needs it's own heading.  And the divider shouldn't be `<li><hr / ><li>` (similarly to the UserMenu) 

Context for this _big_ change:
Previously the `<ProfileMenu />` was encapsulated within the `<ul>` of the `<UserMenu />`, which doesn't allow a heading. So the `<ProfileMenu />` needed to be placed above the  `<ul>` of the `<UserMenu />`. But because the Usermenu was previously merely used as an pass-though component, passing a long a lot of props from `<UserMenu />` to `<ProfileMenu />` it felt logical to split them in to separate components.

This change also optimise other things such as:
1.  Applied correct scaling of the plus-sign ("Add profile") in the profile menu. Happened due to icon restructure.
2. The menu would not close when pressing the profile menu buttons. Happened due to the fly-out menu implementation was incomplete.
3. Adds `aria-current="true"` to the active profile menu button
4. Removed some dead CSS (`.profileButton`)
5. Applied small optimisations for the Profile Edit form.
5a. Logical heading order
5b. Always enabled submit button
5c. Correct `aria-invalid` state by removing the generic form error condition

Ticket: https://videodock.atlassian.net/browse/OTT-979

### CODE REVIEW

**Comment by langemike** [comment]
@ChristiaanScheermeijer noted that the focus outline were missing on some buttons (eg. logout) inside the menu.

That's when I realised `role="button"` was missing on the `<MenuButton />`. This proves that sometimes low hanging fruit can easily be missed. 😅

File: packages/ui-react/src/containers/Profiles/Form.tsx [hunk 1]
```diff
@@ -74,14 +73,13 @@ const Form = ({ initialValues, formHandler, selectedAvatar, showCancelButton = t
         <div className={profileStyles.formFields}>
           {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
           {submitting && <LoadingOverlay />}
-          <h2 className={profileStyles.nameHeading}>{t('name')}</h2>
```
Comment (royschut) [inline packages/ui-react/src/containers/Profiles/Form.tsx:77]: This one seems to be removed completely. Is that on purpose?

File: packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  onButtonClick?: () => void;
+};
+
+const ProfilesMenu = ({ profiles, currentProfile, small, selectingProfile, selectProfile, onButtonClick }: ProfilesMenuProps) => {
+  const { t } = useTranslation('user');
+  const navigate = useNavigate();
+
+  const onProfileSelectClick = (profile: Profile) => {
+    onButtonClick?.();
+    selectProfile?.({ id: profile.id, avatarUrl: profile.avatar_url });
```
Comment (royschut) [inline packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx:28]: We can't make these handlers required?

**Review by royschut — COMMENTED** [review]
Great improvement Mike! You've covered a lot.

I have a few small considerations, and I will have another scan tomorrow morning.

File: packages/ui-react/src/containers/Profiles/Form.tsx (hunk 1 above)
Comment (langemike) [reply packages/ui-react/src/containers/Profiles/Form.tsx:77]: Yes. It was an arbitrary "Name" on the page. You can see it for your self if you visit your current profile edit page on https://svod-inplayer-acc.demo.videodock.com/

File: packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import MenuButton from '../MenuButton/MenuButton';
+import Icon from '../Icon/Icon';
+import ProfileCircle from '../ProfileCircle/ProfileCircle';
+
+type ProfilesMenuProps = {
+  profiles: Profile[];
+  currentProfile?: Profile | null;
+  small?: boolean;
+  selectingProfile: boolean;
+  selectProfile?: UseMutateFunction<unknown, unknown, { id: string; avatarUrl: string }, unknown>;
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx]: I would try using simple props in components without creating awareness of how the parent implements the `selectProfile` prop.

```suggestion
  selectProfile: ({ id: string; avatarUrl: string }) => void;
```

File: packages/ui-react/src/components/Header/Header.tsx
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/Header/Header.tsx:1]: There is a `.profiles` selector in the Header.modules.scss. I think this can be removed.

**Review by ChristiaanScheermeijer — APPROVED** [review]
PR looks good to me, I have minor feedback, but that's not super important.

File: packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx (hunk 2 above)
Comment (langemike) [reply packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx:28]: Based on your feedback, I tried to make the `selectProfile` required. But I ran in to all kind of typescript issues of parent components (related to the `profileData` prop) which I tried to solve. After 1 hour I thought it was enough.

Do you consider this important? Then I suggest looking at it together (tomorrow).

I suggest maybe leaving it as-is and refactor/improve it in the future. I already spent quite some time on this.

File: packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx (hunk 3 above)
Comment (langemike) [reply packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx]: changed!

File: packages/ui-react/src/components/Header/Header.tsx
Comment (langemike) [reply packages/ui-react/src/components/Header/Header.tsx:1]: I removed the `.profile` and `.userMenuButton` selector from Header.modules.scss. They are both unused

File: packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx (hunk 2 above)
Comment (royschut) [reply packages/ui-react/src/components/ProfilesMenu/ProfilesMenu.tsx:28]: Sorry Mike, I should've called this minor, I find the 1 hour investment already a lot. Another time, and thanks for your attempt!

**Review by royschut — APPROVED** [review]

**Pull request merged**